rear-panel/firmware/host_comms_task/freertos_comms_task.cpp

```diff
@@ -0,0 +1,178 @@
+}
+}  // namespace host_comms_control_task
+
+static auto cdc_init_handler() -> uint8_t * {
+    using namespace host_comms_control_task;
+    _local_task.committed_rx_buf_ptr = _local_task.rx_buf.committed()->data();
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    return reinterpret_cast<uint8_t *>(_local_task.committed_rx_buf_ptr);
+}
+
+static auto cdc_deinit_handler() -> void {
+    using namespace host_comms_control_task;
+    _local_task.committed_rx_buf_ptr = _local_task.rx_buf.committed()->data();
+}
+
+/*
+** CDC_Receive is a callback hook invoked from the CDC class internals in an
+**
+** interrupt context. Buf points to the pre-provided rx buf, into which the data
+** from the hardware-isolated USB packet memory area has been copied; Len is a
+** pointer to the length of data.
+**
+** Because the host may send any number of characters in one USB packet - for
+** instance, a host that is using programmatic access to the serial device may
+** send an entire message, while a host that is someone typing into a serial
+** terminal may send one character per packet - we have to accumulate characters
+** somewhere until a full message is assembled. To avoid excessive copying, we
+** do this by changing the exact location of the rx buffer we give the
+** USB infrastructure. The rules are
+**
+** - We always start after a buffer swap with the beginning of the committed
+**   buffer
+** - When we receive a message
+**   - if there's a newline (indicating a complete message), we swap the buffers
+**     and  send the one that just got swapped out to the task for parsing
+**   - if there's not a newline,
+**     - if, after the message we just received, there is not enough space for
+**       an entire packet in the buffer, we swap the buffers and send the
+**       swapped-out one to the task, where it will probably be ignored
+**     - if, after the message we just received, there's enough space in the
+**       buffer, we don't swap the buffers, but advance our read pointer to just
+**       after the message we received
+**
+** Just about every line of this function has a NOLINT annotation. This is bad.
+** But the goal is that this is one of a very few functions like this, and
+** changes to this function require extra scrutiny and testing.
+*/
+
+// NOLINTNEXTLINE(readability-non-const-parameter)
+static auto cdc_rx_handler(uint8_t *Buf, uint32_t *Len) -> uint8_t * {
+    using namespace host_comms_control_task;
+    ssize_t remaining_buffer_count =
+        (_local_task.rx_buf.committed()->data()
+         // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+         + _local_task.rx_buf.committed()->size()) -
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic,cppcoreguidelines-pro-type-reinterpret-cast)
+        reinterpret_cast<char *>(Buf + *Len);
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+    if ((std::find_if(Buf, Buf + *Len,
```

Comment:
idk if we want to do it in this PR (probably not) but we'll need to replace this with the binary messaging handling stuff instead of text mode